src/ameba/rule/style/verbose_block.cr

```diff
@@ -0,0 +1,229 @@
+  #   ExcludeOperators: false
+  #   ExcludeSetters: false
+  #   MaxLineLength: ~
+  #   MaxLength: 50 # use ~ to disable
+  # ```
+  class VerboseBlock < Base
+    properties do
+      description "Identifies usage of collapsible single expression blocks."
+
+      exclude_calls_with_block true
+      exclude_multiple_line_blocks false
+      exclude_prefix_operators true
+      exclude_operators false
+      exclude_setters false
+
+      max_line_length : Int32? = nil # 100
+      max_length : Int32? = 50
+    end
+
+    MSG          = "Use short block notation instead: `%s`"
+    CALL_PATTERN = "%s(%s&.%s)"
+
+    protected def same_location_lines?(a, b)
+      return unless a_location = a.name_location
+      return unless b_location = b.location
+
+      a_location.line_number == b_location.line_number
+    end
+
+    private PREFIX_OPERATORS = {"+", "-"}
+    private OPERATOR_CHARS   =
+      {'[', ']', '!', '=', '>', '<', '~', '+', '-', '*', '/', '%', '^', '|', '&'}
+
+    protected def prefix_operator?(node)
+      node.name.in?(PREFIX_OPERATORS) && node.args.empty?
+    end
+
+    protected def operator?(name)
+      name.each_char do |char|
+        return false unless char.in?(OPERATOR_CHARS)
+      end
+      !name.empty?
+    end
+
+    protected def setter?(name)
+      !name.empty? && name[0].letter? && name.ends_with?('=')
+    end
+
+    protected def valid_length?(code)
+      if max_length = self.max_length
+        return code.size <= max_length
+      end
+      true
+    end
+
+    protected def valid_line_length?(node, code)
+      if max_line_length = self.max_line_length
+        if location = node.name_location
+          final_line_length = location.column_number + code.size
+          return final_line_length <= max_line_length
+        end
+      end
+      true
+    end
+
+    protected def reference_count(node, obj : Crystal::Var)
+      i = 0
+      case node
+      when Crystal::Call
+        i += reference_count(node.obj, obj)
+        i += reference_count(node.block, obj)
+
+        node.args.each do |arg|
+          i += reference_count(arg, obj)
+        end
+        node.named_args.try &.each do |arg|
+          i += reference_count(arg.value, obj)
+        end
+      when Crystal::Block
+        i += reference_count(node.body, obj)
+      when Crystal::Var
+        i += 1 if node == obj
+      end
+      i
+    end
+
+    protected def args_to_s(io : IO, node : Crystal::Call, skip_last_arg = false)
+      node.args.dup.tap do |args|
+        args.pop? if skip_last_arg
+        args.join io, ", "
+        node.named_args.try do |named_args|
+          io << ", " unless args.empty? || named_args.empty?
+          named_args.join io, ", " do |arg, inner_io|
+            inner_io << arg.name << ": " << arg.value
+          end
+        end
+      end
+    end
+
+    protected def args_to_s(node : Crystal::Call, skip_last_arg = false)
+      String.build { |io| args_to_s(io, node, skip_last_arg) }
+    end
```

Comment:
this method seems to be unused.

Reply:
That's correct, I've added it for the "symmetry", but in the end I was also thinking of getting rid of it ;)